File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        lookup = Counter(s)
        buckets = defaultdict(list)
        n = len(s)
        result = ""

        for char, count in lookup.items():
            buckets[count].append(char)

        for i in range(n, 0 , -1):
            for char in buckets[i]:
                result += char * i


        return result
        

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.js
var frequencySort = function(s) {
    let lookup = {}

    for (let char of s) {
        if (!(char in lookup)) lookup[char] = 0
        lookup[char] += 1
    }    

    let array = []
    

    for (let [key, value] of Object.entries(lookup)) {
        array.push([key, value])
    }

    array.sort((a, b)=> b[1] - a[1])
    let output = ""
    for (let [key, value] of array) {
       output += key.repeat(value)
    }

    return output
};